### PR TITLE
Change around the redirect semantics.

### PR DESCRIPTION
**Pull request by aisola**
Fixes MasoniteFramework/core#114
Fixes MasoniteFramework/core#113

### CODE REVIEW

**Comment by josephmancuso** [comment]
This looks awesome. Let me pull it in and run some user tests on it. Hey what do you think of this:

1. Change the redirect_to back to redirectTo.
2. Merge this into 1.6 branch to fix the bug. 
3. Make a new PR and change it back to redirect_to
4. merge it into develop for the 2.0 branch?

In the mean time I'll test this out

**Comment by josephmancuso** [comment]
I like it I just found 1 bug. 

in StartResponse there are these lines:

```python
            self.app.bind('Headers', [
                ('Location', Request.compile_route_to_url())
            ] + Request.get_cookies())
```

Since we made changes to the compile_route method this should I believe change to:

```python
            self.app.bind('Headers', [
                ('Location', Request.redirect_url) # <- Change here
            ] + Request.get_cookies())
```

**Comment by josephmancuso** [comment]
Also, are we using the `request.redirect_route` attribute at all anymore?

**Comment by aisola** [comment]
Oh good catch. I didn't think so, but I left it in there just in case. We aren't using it in any of the rewritten redirects. I forgot to mention it. Should it be removed?

**Comment by josephmancuso** [comment]
we'll need to keep it for 1.6. Let's remove it in the PR for develop

**Comment by aisola** [comment]
Cool fair enough.

**Comment by aisola** [comment]
`StartResponseProvider` has now been fixed.

**Comment by josephmancuso** [comment]
Hm it looks like you merged the develop branch with the 1.6 branch so this PR is also adding the changes to resolving controller constructors. If it too annoying to cherry pick the changes let me know and I'll just merge this into 1.6 anyway and move that from the 2.0 docs to the 1.6 docs.

**Comment by josephmancuso** [comment]
You might actually just be able to `git revert <commit-hash-id>`

**Comment by aisola** [comment]
Okay @josephmancuso, I believe that should fix it.

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
         'masonite.contracts',
         'masonite.helpers',
     ],
-    version='1.6.5',
+    version='1.6.4',
```
Comment (josephmancuso) [inline setup.py]: This should be 1.6.6 so we can release the next bug fix

Comment (aisola) [reply setup.py]: Oh shoot! Sorry! I'll get that done.

**Comment by aisola** [comment]
Bumped

**Comment by josephmancuso** [comment]
🥇 🥇

**Pull request merged**